catalog/spatial/registry/registry-report-action-provider/src/test/java/org/codice/ddf/registry/report/action/provider/TestRegistryMetacardActionProvider.java

```diff
@@ -0,0 +1,397 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.registry.report.action.provider;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.core.IsNot.not;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.codice.ddf.registry.common.RegistryConstants;
+import org.junit.Test;
+
+import ddf.action.Action;
+import ddf.catalog.data.impl.MetacardImpl;
+
+public class TestRegistryMetacardActionProvider {
+
+    protected static final String SAMPLE_PATH = "/registries/";
+
+    protected static final String SAMPLE_SERVICES_ROOT = "/services";
+
+    protected static final String SAMPLE_PROTOCOL = "http://";
+
+    protected static final String SAMPLE_SECURE_PROTOCOL = "https://";
+
+    protected static final String SAMPLE_PORT = "8181";
+
+    protected static final String SAMPLE_SECURE_PORT = "8993";
+
+    protected static final String SAMPLE_IP = "192.168.1.1";
+
+    protected static final String SAMPLE_ID = "abcdef1234567890abdcef1234567890";
+
+    protected static final String ACTION_PROVIDER_ID = "catalog.view.metacard";
+
+    protected static final Set<String> SAMPLE_REGISTRY_TAGS = new HashSet<>(Arrays.asList(
+            RegistryConstants.REGISTRY_TAG));
+
+    protected static final String PATH_AND_FORMAT = "/report.html";
+
+    protected static final String SAMPLE_SOURCE_ID = "ddf.distribution";
+
+    protected static final String SOURCE_ID_QUERY_PARAM = "?sourceId=";
+
+    @Test
+    public void testMetacardNull() {
+        assertThat(new RegistryMetacardActionProvider(ACTION_PROVIDER_ID).getActions(null),
+                hasSize(0));
+    }
+
+    @Test
+    public void testUriSyntaxException() {
+
+        MetacardImpl metacard = new MetacardImpl();
```

Comment:
You should consider something like adding `private Metacard getSampleMetacard()`. Basically you could create a `MetacardImpl` and set the sample ID and/or registry tag in that method. It would reduce the duplication as most of these test methods start with the same logic for building a sample Metacard.